File: 643. Maximum Average Subarray I.php
class Solution {
    function findMaxAverage($nums, $k) {
    $avg = -INF;
    $sum = 0;
    $rigth = 0;
    $left = 0;
    while ($rigth < count($nums)) {
        $sum += $nums[$rigth];
        if (($rigth - $left) + 1 == $k) {
            $avg = max($avg, $sum / $k);
            $sum -= $nums[$left];
            $left++;
        }
        $rigth++;
    }
    return $avg;  
    }
}

File: 643. Maximum Average Subarray I.kt
class Solution {
fun findMaxAverage(nums: IntArray, k: Int): Double {
var avg = Double.NEGATIVE_INFINITY
var sum = 0
var right = 0
var left = 0
while (right < nums.size) {
    sum += nums[right]
    if ((right - left) + 1 == k) {
        avg = max(avg, sum.toDouble() / k.toDouble())
        sum -= nums[left]
        left++
    }
    right++
}
return avg
}
}
